muikku-core-plugins-persistence/src/main/java/fi/otavanopisto/muikku/plugins/chat/model/UserChatSettings.java

```diff
@@ -0,0 +1,63 @@
+package fi.otavanopisto.muikku.plugins.chat.model;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.validation.constraints.NotNull;
+
+import org.hibernate.validator.constraints.NotEmpty;
+
+@Entity
+public class UserChatSettings {
+  public UserChatSettings() {
+    
+  }
+  
+  public UserChatSettings(String userIdentifier, UserChatVisibility visibility) {
+    super();
+    this.userIdentifier = userIdentifier;
+    this.visibility = visibility;
+  }
+
+  public Long getId() {
+    return id;
+  }
+
+  public void setId(Long id) {
```

Comment:
Hibernate has full control of `id` so never add a setter for it so that no one accidentally uses it.